### DISCUSSION TITLE
Add face model pod when developing in windows

### DISCUSSION THREAD

**Discussion by itamardo** [firebase/flutterfire#3938] [Q&A]
Hi
I'm developing an app on Windows 10 and using firebase_ml_vision for face detection.
This works perfectly when running on an Android device.
For creating the iOS build I use Codemagic since I don't have a mac.
The app compiles and works just fine on an iPhone, including other firebase plugins, and I have changed the target platform to 9 to pass compilation.
However, I am unable to detect faces on iOS because I haven't added the face model pod.
I tried to add the pod using the instructions for installing Ruby and Cocoapods and was able to install and verify the toolchain. I created a podfile in the ios folder but when I issue pod update or pod install It fails, and it seems to be because of the file path limit of windows.
I removed the windows limit of the path length in my group policy but this still happens, so I guess Cocoapods doesn't use the windows long path capabilities.

This is the error I see:
`Errno::ENOENT - No such file or directory @ dir_s_mkdir - C:/Users/username/Library/Caches/CocoaPods/Pods/Release/Firebase/6.34.0-c23a3/FirebaseInAppMessaging/Tests/Integration/FunctionalTestApp/ExternalAppExample/fiam-external-ios-testing-app/fiam-external-ios-testing-app/Assets.xcassets/AppIcon.appiconset`

When I go to the "Assets.xxassets" and try to create any folder, I get an OS error that the path is too long so that's probably the reason this fails.
If I had some way to force Cocoapods not to use such a long base directory that would have probably been fine but I couldn't find such an option.

I tried other shells like PWSH (core, v7.0.3) and git bash.
Is there some other way I can add the face model for iOS on windows?

here is my podfile

```
# Uncomment the next line to define a global platform for your project
platform :ios, '9.0'
# source 'https://cocoapods-cdn.netlify.app/'
source 'https://github.com/CocoaPods/Specs.git'
# project './Runner.xcodeproj'

target 'Runner' do
  # Comment the next line if you don't want to use dynamic frameworks
  use_frameworks!

  # Pods for Runner
  # pod 'Firebase/Core'
  # pod 'Firebase/MLVision'
  #pod 'Firebase', '6.25.0'
  #pod 'Firebase/MLVisionFaceModel', '6.25.0'
  pod 'Firebase'
  pod 'Firebase/MLVisionFaceModel'
end
```

**Comment by itamardo** [comment]
update: although a bit crude, I opened cocoapods/config.rb and changed cache_root to the constant 'c:/x' after creating that folder. After that, pod update went through.
now i'm facing this, which is unrelated to the path issue:

`[!] CocoaPods did not set the base configuration of your project because your project already has a custom config set. In order for CocoaPods i
ntegration to work at all, please either set the base configurations of the target `Runner` to `Target Support Files/Pods-Runner/Pods-Runner.de
bug.xcconfig` or include the `Target Support Files/Pods-Runner/Pods-Runner.debug.xcconfig` in your build configuration (`Flutter/Debug.xcconfig
`).                                                                                                                                            
                                                                                                                                               
[!] CocoaPods did not set the base configuration of your project because your project already has a custom config set. In order for CocoaPods i
ntegration to work at all, please either set the base configurations of the target `Runner` to `Target Support Files/Pods-Runner/Pods-Runner.re
lease.xcconfig` or include the `Target Support Files/Pods-Runner/Pods-Runner.release.xcconfig` in your build configuration (`Flutter/Release.xc
config`).                                                                                                                                      
                                                                                                                                               
[!] CocoaPods did not set the base configuration of your project because your project already has a custom config set. In order for CocoaPods i
ntegration to work at all, please either set the base configurations of the target `Runner` to `Target Support Files/Pods-Runner/Pods-Runner.pr
ofile.xcconfig` or include the `Target Support Files/Pods-Runner/Pods-Runner.profile.xcconfig` in your build configuration (`Flutter/Release.xc
config`).                                                                                                                                      `

**Comment by davidair** [comment]
Could this be related to https://github.com/flutter/flutter/issues/66222? I wonder if you could temporarily borrow a Mac (or rent a VM online) and diff the Xcode project you get there from the one you get on Windows.

**Comment by itamardo** [comment]
That would probably be a solution, yes, but I am trying to avoid that for now. I really want the approach of cross-platform and no-need for mac would work end-to-end using flutter and codemagic (and, hopefully, little platform-specific knowledge).
Anyway, I finally got everything to work now using that route and I'll post how later, either here or in a new discussion with a link here.

**Comment by itamardo** [comment]
Here's what I did to finally get face detection working, using only my windows host and codemagic to build the iOS version.

1. Follow all the instructions in the [firebase_ml_vision ](https://pub.dev/packages/firebase_ml_vision) plugin, regarding Android and get it to work on Android.
2. Instead of creating your own Podfile, go into the firebase_ml_vision example repo and take the Podfile from there. Put it under the ios folder just like in the example.
3. In that Podfile, change

```
post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings['ENABLE_BITCODE'] = 'NO'
    end
  end
end
```

to

```
post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings['ENABLE_BITCODE'] = 'NO'
      config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '9.0'
    end
  end
end
```

this is based on suggestion from [Coeur](https://github.com/CocoaPods/CocoaPods/issues/7314) and will cause codemagic stop complaining about the deployment target. Turns out just setting the platform in the top of the podfile is not enough.

4. at the bottom of the podfile, comment out or delete all the models that don't interest you
5. make sure all instances of IPHONEOS_DEPLOYMENT_TARGET in your project are at 9.0 and not 8.0
6. in ios/flutter change debug and release.xcconfig according to the example in firebase_ml_vision. basically this means adding an include to the Pods folder (that will be later created in codemagic) before the current include.
7. in codemagic, if you are building a debug build, you will get a public api warning from apple. select a release build in your codemagic app build options to avoid that warning.

That's it. Turns out that having the **correct** pod file is enough and there is no need to run cocoapods at all. Codemagic runs its own cocoapods. If you had previously run it, then erase the podfile.lock file and push that deletion before building with codemagic.
